src/lib/FieldTypeProcessorRegistry.php

```diff
@@ -32,10 +32,22 @@ public function __construct(array $processors = [])
      * Registers $processor for $fieldTypeIdentifier.
      *
      * @param string $fieldTypeIdentifier
-     * @param \EzSystems\EzPlatformRest\FieldTypeProcessor $processor
+     * @param \EzSystems\EzPlatformRest\FieldTypeProcessorInterface $processor
      */
-    public function registerProcessor($fieldTypeIdentifier, FieldTypeProcessor $processor)
+    public function registerProcessor($fieldTypeIdentifier, /*FieldTypeProcessorInterface*/ $processor)
     {
+        // The FieldTypeProcessorInterface was a late addition to the FieldTypeProcessor class.
+        // The type validation is done this way to insure BC, while warning users of the need to fix their code.
+        if (! $processor instanceof FieldTypeProcessorInterface) {
+            $errorMessage = 'Argument 2 passed to ' . get_class($this) . '::registerProcessor() must be an' .
+                ' instance of EzSystems\EzPlatformRest\FieldTypeProcessorInterface, instance of ' . get_class($processor) .
+                ' given';
+            if ($processor instanceof FieldTypeProcessor) {
+                trigger_error($errorMessage, E_USER_DEPRECATED);
+            } else {
+                 throw new \TypeError($errorMessage);
+            }
```

Comment:
Messages should be different and `trigger_error` should probably be silent, similarly to how it was done for Symfony Event Dispatcher when they were working with BC:

https://github.com/symfony/symfony/blob/91121acf4bc276620e765310357c79fb6c7c0b73/src/Symfony/Component/EventDispatcher/EventDispatcher.php#L58-L63

Deprecation notice should provide instructions about how to solve them (which would be add interface to `$processor`)

Ping @alongosz is this ok for you?

Reply:
(see my last comment above)